deepchem/feat/molecule_featurizers/one_hot_featurizer.py

```diff
@@ -32,21 +32,26 @@ class OneHotFeaturizer(Featurizer):
   It does not need RDKit to be installed to work with arbitrary strings.
   """
 
-  def __init__(self, charset: List[str] = ZINC_CHARSET, max_length: int = 100):
+  def __init__(self,
+               charset: List[str] = ZINC_CHARSET,
+               max_length: Optional[int] = 100):
     """Initialize featurizer.
 
     Parameters
     ----------
-    charset: List[str], optional (default ZINC_CHARSET)
+    charset: List[str] (default ZINC_CHARSET)
       A list of strings, where each string is length 1 and unique.
-    max_length: int, optional (default 100)
-      The max length for SMILES string. If the length of SMILES string is
-      shorter than max_length, the SMILES is padded using space.
+    max_length: Optional[int], optional (default 100)
+      The max length for string. If the length of string is shorter than
+      max_length, the string is padded using space.
     """
     if len(charset) != len(set(charset)):
       raise ValueError("All values in charset must be unique.")
     self.charset = charset
-    self.max_length = max_length
+    if max_length is not None:
```

Comment:
Why would we want `self.max_length` to be None? Does that specify arbitrary length strings are allowed? If so, we should document that above in the docstring

Reply:
Yes, it specifies arbitrary length strings allowed.

That makes sense.